### ISSUE TITLE
[BUG] React or Web SDK not working with NextJS when building project

### ISSUE THREAD

**Issue by juanparadox** [open-feature/js-sdk#692]
### Observed behavior

I am trying to use an internal tool that uses the React SDK and Web SDK as dependencies along with the LaunchDarkly client provider. Running the app in the local dev server that comes out of the box with NextJS works as expected, but once I try to build the project I run into this:

```
TypeError: Cannot read properties of null (reading 'Symbol(@openfeature/web-sdk/api)')
    at io.getInstance ([redacted]/nextjs/.next/server/app/page.js:18:171732)
```

Weird thing (or maybe it's not weird?) is that the code is literally trying to read from `null` as it seems that is what `globalThis` is being set to based on this snippet of built code from the next server within `page.js`

```
it = Symbol.for('@openfeature/web-sdk/api'),
          io = class extends null {
            constructor() {
              super('client'),
                (this._events = new o9()),
                (this._defaultProvider = o8),
                (this._createEventEmitter = () => new o9());
            }
            static getInstance() {
              let e = null[it];
              if (e) return e;
              let t = new io();
              return (null[it] = t), t;
            }
```

Does the web or React SDK support NextJS?

### Expected Behavior

No property read error to occur.

### Steps to reproduce

1. Create a package that exports a React component which uses the react SDK's Provider component and hooks.
2. Import said package into a NextJS application
3. Try to build the NextJS app

**Comment by beeme1mr** [comment]
Hey @juanparadox, the React SDK is a work in progress, but @toddbaert is actively working on it. We haven't tested it with NextJS yet but we plan on supporting it.

It looks like the Spotify team was able to get OpenFeature [working in NextJS](https://github.com/spotify/confidence-openfeature-provider-js/tree/main/examples) but they're using their own React implementation. Perhaps there's an issue in our React SDK implementation. FYI @toddbaert

What version of NextJS are you using? Did you customize the bundling configuration?

**Comment by juanparadox** [comment]
@beeme1mr To clarify, it works fine when imported directly into the project and used as directed, just not when used with a library that uses the OpenFeature SDKs (sorry if that wasn't immediately apparent). I am getting this error with version `13.3.1` and `latest` (currently at `14.0.3`). No bundling configuration has been modified.

@weyert Yes, I am using the app router.

**Comment by juanparadox** [comment]
I'm going to try removing the React SDK from the library to see if the build issue goes away. Will update shortly.

**Comment by beeme1mr** [comment]
Interesting, thanks. The `extends null` part looks very odd to me. Would you happen to have an example repo available?

**Comment by juanparadox** [comment]
@beeme1mr I wish I did, but it's a private repo and the library is also private 😭 If removing the React SDK doesn't solve the issue (I will know shortly), I can try and recreate the NextJS app and a small library repo to provide for replication.

**Comment by juanparadox** [comment]
Okay so I fixed the issue and it's existence is due to the `web-sdk` being built for server side use instead of client side use. I was importing a file like this:

```
import { OpenFeature } from '@openfeature/web-sdk';
import { LaunchDarklyClientProvider } from '@openfeature/launchdarkly-client-provider';

const launchDarklyClientProvider = new LaunchDarklyClientProvider('1234', {});

function initializeOpenFeature() {
  return OpenFeature.setProviderAndWait(launchDarklyClientProvider);
}

export default initializeOpenFeature;
```

which was being imported into a NextJS file being used server side. Moving the initialization into a client side component fixed the build issue.

It feels like when implementing the `web-sdk` into a NextJS app we need to make sure it's used in a client side component (which feels obvious in hindsight), but I'm just not sure how one might try to initialize/set a provider server side for use on the client (not sure if that's even a supported feature anyway).

TLDR: I tried to initialize the provider server side with NextJS (not in a `use client` component) and it was causing build issues. The issue is not apparent when running the local NextJS dev server, only when building the app.

**Comment by beeme1mr** [comment]
Glad you were able to figure it out. NextJS definitely starts to blur the lines between the server and the client. We'll have to keep that in mind as we build out our NextJS support.

Since you have it working, I'm going to close this issue.
